Src/Pc/CompilerCore/Backend/Coyote/CoyoteCodeGenerator.cs

```diff
@@ -775,6 +757,11 @@ private void WriteStmt(CompilationContext context, StringWriter output, IPStmt s
                     break;
 
                 case FunCallStmt funCallStmt:
+                    if (funCallStmt.Function.CanChangeState == true || funCallStmt.Function.CanRaiseEvent == true)
```

Comment:
@pdeligia I just realized this thing would not work out. Consider the function:
```
foo() {
if($) 
  goto S;
}
```
In other words, we will always early terminate on a function call that can change state or raise an event in its function body.

Reply:
Good catch! This should be now fixed.